src/main/java/at/hannibal2/skyhanni/features/fishing/SeaCreatureFeatures.kt

```diff
@@ -81,14 +83,20 @@ object SeaCreatureFeatures {
 
     @HandleEvent(onlyOnSkyblock = true)
     fun onSeaCreatureFish(event: SeaCreatureFishEvent) {
-        if (!config.alertOwnCatches) return
+        if (!config.alertOwnCatches && !config.announceRare) return
 
         if (event.seaCreature.rare) {
-            val text = if (config.creatureName) "${event.seaCreature.displayName}!"
-            else "${event.seaCreature.rarity.chatColorCode}RARE CATCH!"
-            TitleManager.sendTitle(text, height = 2.8, fontSize = 7f)
-            if (config.playSound) SoundUtils.playBeepSound()
-            lastRareCatch = SimpleTimeMark.now()
+            if (config.alertOwnCatches) {
+                val text = if (config.creatureName) "${event.seaCreature.displayName}!"
+                else "${event.seaCreature.rarity.chatColorCode}RARE CATCH!"
+                TitleManager.sendTitle(text, height = 2.8, fontSize = 7f)
+                if (config.playSound) SoundUtils.playBeepSound()
+                lastRareCatch = SimpleTimeMark.now()
+            }
+            if (config.announceRare && PartyApi.partyMembers.size > 0) {
+                if (config.creatureName) HypixelCommands.partyChat("I caught a ${event.seaCreature.name}!")
```

Comment:
i dont think we should use the creatureName config option in the new feature at all. there is no point in sending a chat message telling that you found "something rare" to others without telling them what you caught as well. the question that comes to mind is "why would you say that in party chat manually?" i dont see the need for that.

Reply:
Also please add support for dynamic "i caught a/an" depending on the sea creature name. maybe we even have a utils function for this in string utils already not sure.